src/dbnode/network/server/tchannelthrift/node/service.go

```diff
@@ -453,7 +455,9 @@ func (s *service) Query(tctx thrift.Context, req *rpc.QueryRequest) (*rpc.QueryR
 	}
 	defer s.readRPCCompleted()
 
-	ctx, sp, sampled := tchannelthrift.Context(tctx).StartSampledTraceSpan(tracepoint.Query)
+	ctx := tchannelthrift.Context(tctx)
+	ctx = addSourceToContext(ctx, req.Source)
```

Comment:
Can we actually use this opportunity to make a single method here that combines the unwrapping from tchannelthrift context to real context?

```
func contextWithSource(tctx thrift.Context) context.Context { }
```